### PR TITLE
feat:replace md5 checksum

### PR DESCRIPTION
**Pull request by murali-shris**
**- What I did**
- at_commons changes to replace md5 with sha256/512
**- How I did it**
- introduced new object PublicKeyHash which has the hash of public key and hashing algo used.
- added custom object PublicKeyHash to at_key.dart
- changes to toJson and toProtocolFragment methods in metadata
- deprecated pubKeyCS in at_key.dart
- added unit tests

**- How to verify it**
- unit tests should pass

### CODE REVIEW

File: packages/at_commons/lib/src/keystore/public_key_hash.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/// Represents hash of an atsign's public encryption key and the hashing algorithm used
+class PublicKeyHash {
+  String? hash;
+  PublicKeyHashingAlgo? publicKeyHashingAlgo;
```
Comment (gkc) [inline packages/at_commons/lib/src/keystore/public_key_hash.dart]: Why are these nullable?

Comment (murali-shris) [reply packages/at_commons/lib/src/keystore/public_key_hash.dart]: field should not be nullable. I have removed nullable fields

**Review by gkc — APPROVED** [review]

**Comment by gkc** [comment]
@murali-shris pubKeyCS is marked as deprecated but pubKeyHash is not being used in any other package that I can find. What's the plan here? I think to prevent confusion we need to remove the deprecation annotation until at_server and at_client_sdk fully support pubKeyHash. I'll make a PR to that effect. (PR [here](https://github.com/atsign-foundation/at_libraries/pull/615) was merged a while back)

**Comment by gkc** [comment]
@murali-shris Please see above comment; can you create a ticket to uptake pubKeyHash in (a) atServer (b) at_client_sdk (c) at_c_sdk

**Comment by murali-shris** [comment]
> @murali-shris Please see above comment; can you create a ticket to uptake pubKeyHash in (a) atServer (b) at_client_sdk (c) at_c_sdk
https://github.com/atsign-foundation/at_server/issues/2121
https://github.com/atsign-foundation/at_client_sdk/issues/1417
https://github.com/atsign-foundation/at_c/issues/424

**Pull request merged**